tests/api/api/test_functions.py

```diff
@@ -78,6 +78,57 @@ def test_build_status_pod_not_found(
     assert response.status_code == HTTPStatus.NOT_FOUND.value
 
 
+@pytest.mark.asyncio
+async def test_list_functions_with_hash_key_versioned(
+    db: sqlalchemy.orm.Session, async_client: httpx.AsyncClient
+):
+    await tests.api.api.utils.create_project_async(async_client, PROJECT)
+
+    function = {
+        "kind": "job",
+        "metadata": {
+            "name": "function-name",
+            "project": "project-name",
+            "tag": "some-tag",
+        },
+        "spec": {"image": "mlrun/mlrun"},
+    }
+
+    function2 = {
+        "kind": "job",
+        "metadata": {
+            "name": "function-name",
+            "project": "project-name",
+            "tag": "some-tag2",
+        },
+    }
+
+    post_function1_response = await async_client.post(
+        f"func/{function['metadata']['project']}/"
+        f"{function['metadata']['name']}?tag={function['metadata']['tag']}&versioned={True}",
+        json=function,
+    )
```

Comment:
IMO it would be cleaner to have the tag name and function name of each function in a parameter and pass it along the functions constructions and in the API requests instead pulling keys from the dictionaries.